### ISSUE TITLE
Conflict between gatsby-source-prismic and gatsby-source-filesystem - Breaks Build

### ISSUE THREAD

**Issue by LiamJSP** [prismicio/prismic-gatsby#530]
### Versions

- gatsby-source-prismic: 5.3.0
- gatsby-source-filesystem: 5.9.0
- gatsby: 5.9.0
- node: v18.16.0

### Steps to reproduce
1. Clone this repo: https://github.com/RedCloverSoftware/redclover-website-gatsbyjs.git
2. yarn install
3. yarn upgrade --latest gatsby && yarn upgrade --latest gatsby-source-prismic && yarn upgrade --latest gatsby-source-filesystem
4. Update your global gatsby-cli if not at latest.
5. yarn build
### What is expected?
A successful build
### What is actually happening?
After upgrading to the latest version of gatsby and gatsby-source-prismic, the site can no longer be built, due to a conflict between the plugin gatsby-source-prismic, and the official gatsby plugin gatsby-source-filesystem (used by gatsby-plugin-image). This is the specific error:

**Error: The plugin "gatsby-source-prismic" created a node of a type owned by another plugin.
        The node type "File" is owned by "gatsby-source-filesystem".
        If you copy and pasted code from elsewhere, you'll need to pick a new type name
        for your new node(s).**

[Full build log with error](https://github.com/prismicio/prismic-gatsby/files/11344722/build_log.txt)

Reverting to the previous version of just gatsby-source-prismic that produced working builds for me (5.2.9) did not resolve this error, it appears that the latest version of Gatsby 5.x (specifically 5.9.0) is involved in this error, as this does not happen when using gatsby 4.21.1 and gatsby-source-prismic 5.2.9. Is gatsby-source-prismic not compatible with Gatsby 5.x at the moment? Or is this some sort of odd dependency conflict particular to my repo? I have tried a build with all top-level deps set to pull latest versions ("*"), no change. Appreciate any assistance or thoughts!

**Comment by LiamJSP** [comment]
Sooo I just tried to create a prismic gatsby starter to test whether this issue is particular to my repo only or truly an issue with the latest release of Gatsby and gatsby-source-prismic, and it appears the prismic gatsby starter is also broken? Running through it with a new account, I get the following error preventing creation of a new starter:

"creating prismic repository... Invalid custom types format".

Obviously this would necessitate it's own issue but I would really appreciate a sanity check that these aren't issues just particular to me, as from my end its looking like gatsby support is thoroughly broken at the current 5.x version for Prismic...

**Comment by LiamJSP** [comment]
I tried to setup a prismic gatsby starter [following the official prismic guide](https://prismic.io/docs/starters-gatsby) just to verify whether this issue was unique to my site's repo, and the starter setup also seems to be broken at the current version ("creating prismic repository... invalid custom types format")... can someone do a sanity check for me, is Gatsby 5.x and Prismic currently completely broken? The starter attempts to use Gatsby 5.x so it seems like it should be?

**Comment by angeloashmore** [comment]
Hi @LiamJSP, just wanted to let you know I’m aware of the issue you reported and will take a look at it as soon as possible. I can currently working on Prismic’s Next.js integration and will be revisiting Gatsby right after.

In the meantime, I recommend using the latest version of Gatsby that works with `gatsby-source-prismic`. The plugin works with Gatsby 5, but it’s possible some breaking change was published in the last east version of Gatsby.

**Comment by LiamJSP** [comment]
@angeloashmore Thanks for the heads up! I'll continue to do some digging and add any helpful info I can find or if I'm lucky open a PR :)

**Comment by LiamJSP** [comment]
Small update, the prismic CLI works fine on my macbook, so not sure if it's a global dependency issue with my windows workstation. Just letting you know we can exclude it from this issue, it's just the odd conflict between gatsby-source-prismic and gatsby-source-filesystem, I have an update on this too:

After sucessfully cloning the gatsby prismic starter and upgrading it's Gatsby and all plugins to latest, there was no issue. So this suggests an issue with my repo, but I'm just not sure where to look - the two plugins were working fine prior to upgrading, and the error thrown leaves no room to imaging any other cause ("Error: The plugin "gatsby-source-prismic" created a node of a type owned by another plugin."). Any thoughts where I should be digging for this issue? My gatsby-config.js appears fine, here is the two relevant parts for both plugins:

{
resolve: gatsby-source-filesystem,
options: {
name: images,
path: path.resolve(__dirname, 'src', 'images'),
},
},
{
resolve: gatsby-source-prismic,
options: {
repositoryName: process.env.GATSBY_PRISMIC_REPO_NAME,
accessToken: process.env.PRISMIC_ACCESS_TOKEN,
customTypesApiToken: process.env.PRISMIC_CUSTOM_TYPES_API_TOKEN,
shouldDownloadFiles: () => true,
},
},

Let me know if you're aware of anything that could cause a duplicate node type error after upgrading, only in one particular repo! I'll keep looking too

**Comment by niklaspesch** [comment]
I stumbled across the same issue and can say it's due to shouldDownloadFiles, if you comment that out, the dev server starts up normally. But then the problem is that I can no longer get pictures via localFile. Does this help you further @angeloashmore?

**Comment by maheshsamudra** [comment]
I'm having the same issue with `gatsby-source-filesystem@5.10`, `gatsby@5.10` and `gatsby-source-prismic@5.3.1` on Mac. The issue disappears with `shouldDownloadFiles: false`, but that's not an option for my project.

**Comment by LiamJSP** [comment]
Popping back in to add that I've since found that this issue affected the entire Gatsby source plugin ecosystem. Almost every source plugin that utilized a File node type was broken by Gatsby's changes in recent versions. You can find similar issues for Strapi (https://forum.strapi.io/t/is-gatsby-source-strapi-currently-broken/28147) , DirectUs (https://github.com/gatsby-uc/plugins/issues/426) etc. Despite the claims of the Gatsby error, Prismic obviously didn't copy-paste code, and from my investigation it looks like they're using a link type to tie-in to the vanilla "File" node to allow for consistency in GQL queries (refer to files as localFile). This should and was working prior to the recent gatsby changes.

I experimented a bit with various bandaid fixes I found in different plugin repos (moved gatsby-source-prismic above gatsby-source-filesystem in gatsby-config.js, attempted to rename the internal.type name for File within gatsby-source-prismic, etc). but didn't have any luck.

**TL;DR It's a gatsby bug/breaking change. No official update or guidance I could find at this time.**

**Comment by LiamJSP** [comment]
A member of the Gatsby team confirmed they're aware and looking into a fix!

**Comment by LekoArts** [comment]
Please see https://github.com/gatsbyjs/gatsby/issues/38117#issuecomment-1592092933 and give it a try :)
